File: apps/api/WherezIt.Domain/Entities/Container.cs
namespace WherezIt.Domain.Entities;

public class Container
{
    public Guid Id { get; set; }
    public Guid WorkspaceId { get; set; }
    public Guid StorageNodeId { get; set; }
    public int BoxNumber { get; set; }
    public string? Name { get; set; }
    public string? Description { get; set; }
    public bool IsArchived { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Workspace Workspace { get; set; } = null!;
    public StorageNode StorageNode { get; set; } = null!;
}

File: apps/web/src/features/containers/types/container.ts
export interface Container {
  id: string;
  workspaceId: string;
  storageNodeId: string;
  boxNumber: number;
  boxId: string;
  name?: string | null;
  description?: string | null;
  physicalLabel?: string | null;
  isArchived: boolean;
  destinationStorageNodeId?: string | null;
  isPacked?: boolean;
  movingPriority?: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateContainerRequest {
  storageNodeId: string;
  name?: string;
  description?: string;
  physicalLabel?: string;
  destinationStorageNodeId?: string;
  isPacked?: boolean;
  movingPriority?: string;
}

export interface UpdateContainerRequest {
  name?: string;
  description?: string;
  physicalLabel?: string | null;
  destinationStorageNodeId?: string | null;
  isPacked?: boolean;
  movingPriority?: string | null;
}
